contributing.md

```diff
@@ -1,9 +1,126 @@
 
-We welcome community contributions to the p5.js web editor! You can contribute even if you are not a developer by documenting bugs, proposing new features, and updating documentation.
+Hi! Thanks for wanting to contribute to P5's web editor! You are awesome! 😊
 
-When opening an issue, be sure to clearly describe the issue, task, or feature. If it is a bug, try to include steps to reproduce or other contextual information, such as the browser or operating system. 
+This **web editor** for **p5.js**, a JavaScript library with the goal of making coding accessible to **artists**, **designers**, **educators**, and **beginners**, is used to help develop using p5.js library in web browser and thus makes for a good starting point for working with p5. 
+
+Here are links to all the sections in this document:
+
+<!-- If you change any of the headings in this document, remember to update the table of contents. -->
+
+- [Code of Conduct](#code-of-conduct)
+- [Technologies Used](#technologies-used)
+- [Development Installation](#development-installation)
+- [File Structure](#file-structure)
+- [New Design](#new-design)
+- [How Can I Contribute ?](#how-can-i-contribute?)
+  - [First Timers](#first-timers)
+  - [Want something more challenging](#want-something-more-challenging)
+  - [Feature Enhancement](#feature-enhancement)
+- [Creating a Pull request](#creating-a-pull-request)
+  - - [Tips](#tips)
+
+# Code of Conduct
+
+Please follow the guidelines mentioned at [CODE OF CONDUCT.md](https://github.com/processing/p5.js-web-editor/blob/master/CODE_OF_CONDUCT.md#p5js-code-of-conduct).
+
+# Technologies Used
+
+ > **MERN stack - MongoDB, Express, React/Redux, and Node**. 
+ 
+ - For a reference to the **file structure format** I am using, please look at the [Mern Starter](https://github.com/Hashnode/mern-starter).
+
+ - This project **does not use CSS Modules, but uses Sass**. [BEM guidelines](http://getbem.com/) are followed. 
+ 
+ - For CSS naming conventions, write OOSCSS (Object-Oriented SCSS) with placeholders and mixins, and follow the [7-1 Pattern](https://sass-guidelin.es/#the-7-1-pattern) for Sass.
+
+ - We're using [ES6](http://es6-features.org/) and transpiling to ES5 using [Babel](https://babeljs.io/). 
+
+ - For reference to the JavaScript style guide, see the [Airbnb Style Guide](https://github.com/airbnb/javascript), [React ESLint Plugin](https://github.com/yannickcr/eslint-plugin-react).
+
+ - The ESLint configuration is based on a few popular React/Redux boilerplates. Open to suggestions on this. If in development, **you're getting annoyed with ESLint**, you can **remove** it from `webpack.config.dev.js` in the JavaScript loader, or disable any line from eslint by commenting at the end of the line `// eslint-disable-line`.
+
+# Development Installation
+
+To get started on a patch, first read the instruction from [README.md](https://github.com/processing/p5.js-web-editor#development-installation).
+
+# File Structure
+
+```
+/client
```

Comment:
this could be a separate PR, but it would make sense to also add the file structure for the API/server side stuff too! it's just MVC

Reply:
Yes, it is incomplete for now. I am removing File structure for now and would upload it in another PR.